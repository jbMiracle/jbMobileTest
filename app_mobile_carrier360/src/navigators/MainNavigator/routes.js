import { StackNavigator } from 'react-navigation';
import { mapValues } from 'lodash';
import Perks from '../../screens/Perks/Perks';
import MyLoads from '../../screens/MyLoads/MyLoads';
import Offers from '../../screens/Offers/Offers';
import FindLoads from '../../screens/FindLoads/FindLoads';
import Home from '../../screens/Home/Home';

// The TabNavigator can (should) be given an array of routeNames to explicitly
// determine the order of the tabs.  If not given, the order *could* differ
// from what you intended based on the order of keys in the routes object.
export const orderedRouteNames = [
  'Home',
  'Offers',
  'FindLoads',
  'MyLoads',
  'Perks',
];

export const initialRouteName = 'Home';

const screensWithSubtabs = {
  Offers: true,
  MyLoads: true,
};

const screens = {
  Home,
  Offers,
  FindLoads,
  MyLoads,
  Perks,
};

export const routes = mapValues(screens, createTabStack);

function createTabStack(screen, routeName) {
  if (screensWithSubtabs[routeName]) {
    return screen;
  }
  const stackRoutes = {
    [routeName]: { screen },
  };
  return StackNavigator(stackRoutes, { headerMode: 'none' });
}
