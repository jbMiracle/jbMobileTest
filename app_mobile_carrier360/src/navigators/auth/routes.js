const LoginPendingMessages = {
  carrierDotNotFoundInCms: 'PENDING/CARRIER_DOT_NOT_FOUND_IN_CMS',
  carrierPending: 'PENDING/CARRIER_PENDING',
  rejected: 'PENDING/REJECTED',
  unverifiedUser: 'PENDING/UNVERIFIED_USER',
  waitingOnAdminApproval: 'PENDING/WAITING_ON_ADMIN_APPROVAL',
  waitingOnEmailActivation: 'PENDING/WAITING_ON_EMAIL_ACTIVATION',
};

export default LoginPendingMessages;
