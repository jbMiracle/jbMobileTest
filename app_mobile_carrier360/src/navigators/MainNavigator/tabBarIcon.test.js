import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TabBarIcon from './tabBarIcon';
import { orderedRouteNames } from './routes';

const mockNav = {
  popToTop: () => {},
  navigate: () => {},
};

const testOnRouteName = (routeName) => {
  const navForRoute = {
    ...mockNav,
    state: { routeName },
  };

  it(`should render the tab bar icon for ${routeName} focused`, () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TabBarIcon navigation={ navForRoute } focused />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it(`should render the tab bar icon for ${routeName} unfocused`, () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TabBarIcon navigation={ navForRoute } focused={ false } />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
};

describe('MainNav tabBarIcon component', () => {
  orderedRouteNames.forEach(testOnRouteName);
});
