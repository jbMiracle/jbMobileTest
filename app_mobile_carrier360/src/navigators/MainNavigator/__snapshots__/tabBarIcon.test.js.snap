// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav tabBarIcon component should render the tab bar icon for FindLoads focused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#007DBA"
  id="FindLoads"
  indicator="#007DBA"
  main={true}
  mainIconColor="#EFEFEF"
  name="jbh-Search"
  onPress={[Function]}
  size={25}
  tintColor="#007DBA"
  title="Find Loads"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for FindLoads unfocused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#EFEFEF"
  id="FindLoads"
  indicator="transparent"
  main={true}
  mainIconColor="#939393"
  name="jbh-Search"
  onPress={[Function]}
  size={25}
  tintColor="#939393"
  title="Find Loads"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for Home focused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#EFEFEF"
  id="Home"
  indicator="#007DBA"
  main={false}
  mainIconColor="#007DBA"
  name="jbh-Home"
  onPress={[Function]}
  size={25}
  tintColor="#007DBA"
  title="Home"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for Home unfocused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#FFFFFF"
  id="Home"
  indicator="transparent"
  main={false}
  mainIconColor="#939393"
  name="jbh-Home"
  onPress={[Function]}
  size={25}
  tintColor="#939393"
  title="Home"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for MyLoads focused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#EFEFEF"
  id="MyLoads"
  indicator="#007DBA"
  main={false}
  mainIconColor="#007DBA"
  name="jbh-Trailers"
  onPress={[Function]}
  size={25}
  tintColor="#007DBA"
  title="My Loads"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for MyLoads unfocused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#FFFFFF"
  id="MyLoads"
  indicator="transparent"
  main={false}
  mainIconColor="#939393"
  name="jbh-Trailers"
  onPress={[Function]}
  size={25}
  tintColor="#939393"
  title="My Loads"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for Offers focused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#EFEFEF"
  id="Offers"
  indicator="#007DBA"
  main={false}
  mainIconColor="#007DBA"
  name="jbh-Tag"
  onPress={[Function]}
  size={25}
  tintColor="#007DBA"
  title="Offers"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for Offers unfocused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#FFFFFF"
  id="Offers"
  indicator="transparent"
  main={false}
  mainIconColor="#939393"
  name="jbh-Tag"
  onPress={[Function]}
  size={25}
  tintColor="#939393"
  title="Offers"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for Perks focused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#EFEFEF"
  id="Perks"
  indicator="#007DBA"
  main={false}
  mainIconColor="#007DBA"
  name="jbh-Perks"
  onPress={[Function]}
  size={25}
  tintColor="#007DBA"
  title="Perks"
  virtual={undefined}
/>
`;

exports[`MainNav tabBarIcon component should render the tab bar icon for Perks unfocused 1`] = `
<Styled(C360TabButton)
  backgroundColor="#FFFFFF"
  id="Perks"
  indicator="transparent"
  main={false}
  mainIconColor="#939393"
  name="jbh-Perks"
  onPress={[Function]}
  size={25}
  tintColor="#939393"
  title="Perks"
  virtual={undefined}
/>
`;
